storage/rocksdb/rdb_iterator.cc

```diff
@@ -208,6 +215,18 @@ int Rdb_iterator_base::calc_eq_cond_len(enum ha_rkey_function find_flag,
   return Rdb_key_def::INDEX_NUMBER_SIZE;
 }
 
+int Rdb_iterator_base::iter_status_to_retval(
+    rocksdb::Iterator *it, const std::shared_ptr<Rdb_key_def> kd,
+    int not_found_code) {
+  if (it->Valid()) return HA_EXIT_SUCCESS;
+
+  rocksdb::Status s = it->status();
+  if (s.ok() || s.IsNotFound()) return not_found_code;
+
+  Rdb_transaction *tx = get_tx_from_thd(m_thd);
+  return rdb_tx_set_status_error(tx, s, *kd, m_tbl_def);
+}
+
```

Comment:
The functions are a bit different:
- `is_valid_iterator`: works on a RDB-level iterator, has no THD or MyRocks transaction
- `iter_status_to_retval`: works on a MyRocks-level iterator, has THD, transaction, key definition.

Almost all uses follow the pattern of calling `is_valid_iterator` first and then `iter_status_to_retval`. But there is also `calculate_cardinality_table_scan`, which works with an RDB-level iterator directly.

Unless we find a different way to address that last use, the best I can think of is add a call to is_valid_iterator to `iter_status_to_retval`, perhaps with a rename, so that it's more natural to use in iteration loops.

Reply:
https://github.com/facebook/mysql-5.6/pull/1444